orchestrator/modules/tools/execution/tool_grants.py

```
@@ -0,0 +1,434 @@
+        if ctx.get("board_task_id") is not None:
+            details["board_task_id"] = ctx.get("board_task_id")
+        grant.details = details
+
+        # PRD-193 S5: an ask that fires with nobody watching must not be
+        # silent — announce the fresh pending grant on non-chat lanes (chat
+        # renders the S3 card live in the conversation; don't double-notify).
+        if lane != "chat":
+            _notify_approval_pending(grant, workspace_id)
+
+        return grant
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)